feat: sync v8.32.0

kecrily opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [JS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)


 sync v8.32.0


#### Related Issues

#### Is there anything you'd like reviewers to focus on?

<!-- markdownlint-disable-file MD004 -->

kecrily committed:
docs: add options to check destructuring in no-underscore-dangle

https://github.com/eslint/eslint/pull/16006

kecrily committed:
docs: add options to check destructuring in no-underscore-dangle

https://github.com/eslint/eslint/pull/16006

kecrily committed:
docs: check assignment patterns in no-underscore-dangle

https://github.com/eslint/eslint/pull/16693

kecrily committed:
docs: use inline code for rule options name

https://github.com/eslint/eslint/pull/16768

kecrily committed:
docs: ignore directives for no-fallthrough

https://github.com/eslint/eslint/pull/16757

kecrily committed:
docs: IA Update page URL move

https://github.com/eslint/eslint/pull/16665

kecrily committed:
docs: IA Update page URL move

https://github.com/eslint/eslint/pull/16665

mdjermanovic reviewed (commented):
In `src/extend/custom-rules.md`, line 166, we should remove `.md` from `(./scope-manager-interface.md#scope-interface)`.

(this was included in https://github.com/eslint/eslint/compare/v8.31.0...v8.32.0)

mdjermanovic reviewed (approved):
LGTM when https://github.com/eslint/zh-hans.docs.eslint.org/pull/95#pullrequestreview-1312367047 is addressed. Feel free to merge this how you think it would be best in combination with https://github.com/eslint/zh-hans.docs.eslint.org/pull/94.

kecrily committed:
feat: sync v8.31.0 (#94)

* docs: User Guide Getting Started expansion

* docs: add options to check destructuring in no-underscore-dangle

https://github.com/eslint/eslint/pull/16006

* docs: adjust some words

* docs: Add function call example for no-undefined

https://github.com/eslint/eslint/pull/16712

* docs: check assignment patterns in no-underscore-dangle

https://github.com/eslint/eslint/pull/16693

* update formatters

* Apply suggestions from code review

Co-authored-by: Strek <ssharishkumar@gmail.com>

---------

Co-authored-by: Strek <ssharishkumar@gmail.com>

kecrily committed:
update
